src/battle_script_commands.c

```diff
@@ -16559,3 +16563,18 @@ void BS_ItemRestorePP(void) {
     PREPARE_SPECIES_BUFFER(gBattleTextBuff1, GetMonData(mon, MON_DATA_SPECIES));
     gBattlescriptCurrInstr = cmd->nextInstr;
 }
+
+void BS_TryRevertWeatherForm(void)
+{
+    NATIVE_ARGS();
+    u16 ability = gBattleMons[gBattlerTarget].ability;
+    if (B_WEATHER_FORM_SUPPRESS >= GEN_5
+        && (ability == ABILITY_FORECAST || ability == ABILITY_FLOWER_GIFT)
+        && TryBattleFormChange(gBattlerTarget, FORM_CHANGE_BATTLE_SWITCH)) // revert form
```

Comment:
What do you recommend I use to turn something into its base form? There’s a small chance I do something janky like this in Dynamax that’ll I’ll have to check and fix 👍

Reply:
I think the way to go would be to have ability as optional parameter for `FORM_CHANGE_BATTLE_WEATHER` in the form change tables. If it has it defined, it will require the ability to maintain the form.